cli/tauri.js/bin/tauri.js

```diff
@@ -1,102 +1,100 @@
 #!/usr/bin/env node
 
+const chalk = require('chalk')
+const pkg = require('../package.json')
+const updateNotifier = require('update-notifier')
+
 const cmds = ['help', 'icon', 'deps']
 const rustCliCmds = ['dev', 'build', 'init', 'info']
 
 const cmd = process.argv[2]
-const chalk = require('chalk')
-const pkg = require('../package.json')
-const updateNotifier = require('update-notifier')
 /**
  * @description This is the bootstrapper that in turn calls subsequent
  * Tauri Commands
  *
  * @param {string|array} command
  */
-let skipUpdateNotifier = false
 const tauri = function (command) {
-    if (typeof command === 'object') {
-        // technically we just care about an array
-        command = command[0]
-    }
-
-    if (rustCliCmds.includes(command)) {
-        const { runOnRustCli } = require('../dist/helpers/rust-cli')
-        if (process.argv && !process.env.test) {
-            process.argv.splice(0, 3)
-        }
-        runOnRustCli(
-            command,
-            (process.argv || []).filter((v) => v !== '--no-update-notifier')
-        ).promise.then(() => {
-            if (command === 'init') {
-                const {
-                    installDependencies
-                } = require('../dist/api/dependency-manager')
-                return installDependencies()
-            }
-        })
-    } else {
-        if (
-            !command ||
-            command === '-h' ||
-            command === '--help' ||
-            command === 'help'
-        ) {
-            console.log(chalk.magenta(` 
-      :oooodddoooo;     ;oddl,      ,ol,       ,oc,  ,ldoooooooc,    ,oc,      
-      ';;;cxOx:;;;'    ;xOxxko'     :kx:       lkd,  :xkl;;;;:okx:   lkd,       
-          'dOo'       'oOd;:xkc     :kx:       lkd,  :xx:     ;xkc   lkd,       
-          'dOo'       ckx:  lkx;    :kx:       lkd,  :xx:     :xkc   lkd,       
-          'dOo'      ;xkl   ,dko'   :kx:       lkd,  :xx:.....xko,   lkd,       
-          'dOo'     'oOd,    :xkc   :kx:       lkd,  :xx:,;cokko'    lkd,       
-          'dOo'     ckk:      lkx;  :kx:       lkd,  :xx:    ckkc    lkd,       
-          'dOo'    ;xOl        lko; :xkl;,....;oOd,  :xx:     :xkl'  lkd,       
-          'okl'    'kd'        'xx'  'dxxxddddxxo'   :dd;      ;dxc  'xo'`))
-            console.log(
-                `${chalk.cyan(
-                    'Description'
-                )} \n This is the Tauri CLI \n ${chalk.magenta('Usage')} \n $ tauri ${[
-                    ...rustCliCmds,
-                    ...cmds
-                ].join('|')} \n ${chalk.cyan(
-                    'Options'
-                )} \n --help, -h     Displays this message \n  --version, -v  Displays the Tauri CLI version`
-            )
+  // notifying updates.
+  if (!(process.argv || []).some((arg) => arg === '--no-update-notifier')) {
+    updateNotifier({
+      pkg,
+      updateCheckInterval: 0
+    }).notify()
+  }
```

Comment:
I added the update notification to the beginning, otherwise it won't work because later in the script, there is early returns and process exit

Reply:
'right!